src/trace-span.ts

```diff
@@ -16,51 +16,52 @@
 
 'use strict';
 
-/**
- * Creates a trace span object.
- * @constructor
- */
-function TraceSpan(name, spanId, parentSpanId) {
-  this.name = name;
-  this.parentSpanId = parentSpanId;
-  this.spanId = spanId;
-  this.kind = 'RPC_CLIENT';
-  this.labels = {};
-  this.startTime = (new Date()).toISOString();
-  this.endTime = '';
+export interface TraceSpanLabels {
+  [propName: string]: string;
 }
 
-
 /**
- * Sets or updates a label value.
- * @param {string} key The label key to set.
- * @param {string} value The new value of the label.
+ * Describes a trace span.
  */
-TraceSpan.prototype.setLabel = function(key, value) {
-  this.labels[key] = value;
-};
+export class TraceSpan {
+  public readonly labels: TraceSpanLabels = {};
+  public readonly startTime: string;
+  public endTime: string = '';
+  public kind: string = 'RPC_CLIENT';
```

Comment:
could this be readonly as well?

Reply:
The `TraceAgent` object sets this field to `'RPC_SERVER'` after the object is created. It would be nice to have the `kind` specified upon creation so that it could be `readonly`, but I want to minimize logical changes in this PR.